File: framework/java/src/main/java/soda/web/resp/TextResponse.java
package soda.web.resp;

import java.nio.charset.StandardCharsets;

public class TextResponse extends BaseResponse {
    
    private String text;

    public TextResponse(int httpCode, String content) {
        super(httpCode, "text/plain; charset=utf-8");
        text = content;
    }

    @Override
    public byte[] getBody() {
        return text.getBytes(StandardCharsets.UTF_8);
    }

    public void setContent(String content) {
        text = content;
    }

}

File: framework/scala/src/main/scala/soda/scala/web/resp/TextResponse.scala
package soda.scala.web.resp

import java.nio.charset.StandardCharsets

class TextResponse(httpCode: Int, var content: String)
  extends BaseResponse(httpCode, "text/plain; charset=utf-8") {

  override def getBody: Array[Byte] = content.getBytes(StandardCharsets.UTF_8)

}
